### ISSUE TITLE
S3 Publish not working correctly

### ISSUE THREAD

**Issue by lswith** [aptly-dev/aptly#349]
aptly version: 0.9.6

Steps to reproduce (all done through the api):
1. Create local repo + add .deb
2. Create snap1 from local repo
3. Publish snap1 to S3
4. Remove .deb from local repo
5. Create snap2 from local repo
6. Delete Publish from S3
7. Publish snap1 to S3 

The .deb is included in the Package but the pool directory isn't created again.

**Comment by igurung** [comment]
Hrmm.. We are also getting same error as you describe @Lswith

**Comment by smira** [comment]
@Lswith are you using `-no-lock` or anything special in API mode?

**Comment by Lswith** [comment]
No. Nothing special. It seems to work as expected with the Cli but not the
Api. Also, if you correct it using the Cli, it works as expected until the
same .deb's are taken out of the repo again.

It also works when using the local mode.

**Comment by mnach** [comment]
Have the same problem. It still appear in v1.4.0(master)

**Comment by chicks-net** [comment]
Has this been fixed @lbolla or is the project just ignoring issue that are "too old" to worry about?

**Comment by lbolla** [comment]
mmm Indeed this issue had a relatively recent comment (from 2020). I closed it by mistake!
